pycbc/frame/frame.py

```diff
@@ -702,10 +723,12 @@ def check_valid(self, values, flag=None):
         """
         if self.valid_on_zero:
             valid = values.numpy() == 0
+        elif self.threshold is not None:
+            valid = values.numpy() > self.threshold
         else:
             if flag is None:
                 flag = self.valid_mask
-            valid = numpy.bitwise_and(values.numpy(), flag) == flag
```

Comment:
Why is this change needed?

Reply:
The idq_ok channel should be a binary on/off timeseries of whether to use iDQ's results. It unfortunately gets read in as a float, which causes an error here if not cast to an int.

Reply:
Pretty sure that is a bug (or at least a questionable choice of data formats) on the iDQ side. I suppose it is late/hard to get that changed though?

Reply:
@maxtrevor you can now revert this change, right?

Reply:
Unfortunately not. I changed the code to use the correct idq ok channel but it turns out that channel is being written as a float. Patrick Godwin said that on the iDQ side it is much easier to write frame files with all channels having the same data type and its relatively trivial to cast the channel to an int after reading it. He would prefer not to change this on the idq side right now as review is already ongoing. So I think we are stuck with this casting to int for now at least.

Reply:
I see… *sigh*. Bigger comment coming in the main discussion.